content/blog/2023-09-07-making-connecting-to-localstack-easier/index.md

```diff
@@ -0,0 +1,173 @@
+
+
+## How to use this new feature
+
+For AWS services like Lambda or ECS, we are running your application code in a pre-configured environment.
+
+_For your own containers, there is some configuration required._
+
+Docker allows the configuration of a container DNS resolver.
+This is done by overriding the `/etc/resolv.conf` file inside the container.
+When using the Docker CLI, you can use the `--dns` flag, or the `dns:` entry of a Docker Compose service.
+This flag accepts an IP address to use for resolving domain names.
+To use this flag, your LocalStack container will need to have a known IP address.
+
+### Setting LocalStack as the DNS server using the Docker CLI
+
+When using the CLI, you can use the `--dns` flag to set your application container DNS to the LocalStack container.
+
+```sh
+# start localstack
+localstack start -d
+localstack wait
+
+# get the ip address of the LocalStack container
+docker inspect localstack_main | \
+	jq -r '.[0].NetworkSettings.Networks | to_entries | .[].value.IPAddress'
+# prints 172.17.0.2
+
+# run your application container
+docker run --rm -it --dns 172.17.0.2 <arguments> <image name>
+```
+
+### Setting LocalStack as the DNS server using Docker Compose
+
+When using Docker Compose, you can specify the IP address that the LocalStack container will be assigned by using a user-defined network, and using the `ipam` configuration settings.
+For example:
+
+```yaml
+version: "3.8"
+
+services:
+  localstack:
+    container_name: "${LOCALSTACK_DOCKER_NAME-localstack-main}"
+    image: localstack/localstack
+    ports:
+      # Now only required if you need to access LocalStack from the host
+      - "127.0.0.1:4566:4566"            
+      # Now only required if you need to access LocalStack from the host
+      - "127.0.0.1:4510-4559:4510-4559"
+    environment:
+      - DEBUG=${DEBUG-}
+      - DOCKER_HOST=unix:///var/run/docker.sock
+    volumes:
+      - "${LOCALSTACK_VOLUME_DIR:-./volume}:/var/lib/localstack"
+      - "/var/run/docker.sock:/var/run/docker.sock"
+    networks:
+      ls:
+        # Set the container IP address in the 10.0.2.0/24 subnet
+        ipv4_address: 10.0.2.20
+
+  application:
+    image: ghcr.io/localstack/localstack-docker-debug:main
+    entrypoint: ""
+    command: ["sleep", "infinity"]
+    dns:
+      # Set the DNS server to be the LocalStack continer
+      - 10.0.2.20
+    networks:
+      - ls
+
+networks:
+  ls:
+    ipam:
+      config:
+        # Specify the subnet range for IP address allocation
+        - subnet: 10.0.2.0/24
+```
+
+We hope that with this new functionality available today, accessing LocalStack should be considerably easier.
+By moving the DNS server into LocalStack and configuring AWS compute environments, your Lambda functions, ECS containers, and EC2 instances should already be able to access LocalStack at `localhost.localstack.cloud`.
```

Comment:
Here I also find the usage of "configuring" without the mention of what we configure (even though it can be derived from context) a tad odd. I think a replacement like this would be preferable:
```suggestion
By moving the DNS server into LocalStack and configuring spawned AWS compute environments to use it by default, your Lambda functions, ECS containers, and EC2 instances should already be able to access LocalStack at `localhost.localstack.cloud`.
```